mlir/include/mlir/Target/LLVMIR/Dialect/GENX/GenIntrinsics.h

```diff
@@ -0,0 +1,69 @@
+/*========================== begin_copyright_notice ============================
+
+Copyright (C) 2017-2021 Intel Corporation
+
+SPDX-License-Identifier: MIT
+
+============================= end_copyright_notice ===========================*/
+#pragma once
+
+#include "GenIntrinsicEnum.h"
+
+#include "llvm/ADT/None.h"
+#include "llvm/IR/Function.h"
+
+#include <string>
+#include <vector>
+
+namespace llvm
+{
+
+namespace GenISAIntrinsic
+{
+
+/// Intrinsic::getName(ID) - Return the LLVM name for an intrinsic, such as
+/// "llvm.ppc.altivec.lvx".
+// std::string getName(ID id, ArrayRef<Type*> Tys = None);
+
+
+struct IntrinsicComments
```

Comment:
Actually, this file is directly copied from the IGC generated header.  I was trying to build IGC's GenISAIntrinsic.a directly by adding the corresponding IGC source *.cpp files.  I commented out the unused functions to minimize the build dependency.  I abandoned the effort due to too many dependent files and resort to add the GenISAIntrinsic.a library.  To minimize future updating work, it will be better to stay close to the original IGC generated header.  I am inclining to add the commented out functions back because they are harmless.

Reply:
ok, once we upstream we will need to fix this up.